src/google.py

```diff
@@ -59,14 +65,55 @@ def login():
         
     except ValueError as ex:
         return  jsonify({
-            'error': " invalid id_token"
+            'error': "invalid id_token"
         }), HTTP_400_BAD_REQUEST
     
     
+    issuer = True if os.environ.get('TOKEN_ISSUER') == google_response.get('iss') else False
+    valid_client = True if os.environ.get('CLIENT_ID') == google_response.get('aud') else False
+    token_exp = True if time.time() > int(google_response.get('exp')) else False
+        
+    
+    if not issuer:
+        return jsonify({
+            'error': "bad id_token"
+        }), HTTP_400_BAD_REQUEST
+        
+    
+    if not valid_client:
+        return jsonify({
+            'error': "bad id_token"
+        }), HTTP_400_BAD_REQUEST
+        
+    if token_exp:
```

Comment:
Does this not work?
```
if google_response.get('exp') < time.time():
   ....
```

If google_response.get('exp') returns a unix epoch time and time.time returns unix? ...

Anyway, if comparing directly without converting to int still does not work, just add a little comment so someone else reading it understands